console-framework-client-api/src/main/java/io/axoniq/console/framework/api/aggregateApi.kt

```diff
@@ -0,0 +1,38 @@
+package io.axoniq.console.framework.api
+
+import java.time.Instant
+
+data class DomainEventsResult(
+        val aggregateId: String,
+        val aggregateType: String,
+        val domainEvents: List<DomainEvent>,
+        val page: Int,
+        val pageSize: Int,
+        val totalCount: Long,
+)
+
+data class DomainEvent(
+        val sequenceNumber: Long,
+        val timestamp: Instant,
+        val payloadType: String,
+        val payload: String?
+)
+
+data class DomainEventsQuery(
+        val aggregateId: String,
+        val page: Int = 0,
+        val pageSize: Int = 10,
+)
+
+data class AggregateSnapshotResult(
+        val type: String,
+        val aggregateId: String,
+        val maxSequenceNumber: Long = 0,
+        val snapshot: String,
+)
+
+data class AggregateSnapshotQuery(
```

Comment:
Like we discussed, we should avoid the word Snapshot for this functionality, as it's not persisted.

I'd say something like `AggregateStateAtSequence`/`AggregateStateAtIndex`